src/utils/get-host-app-url.ts

```diff
@@ -0,0 +1,36 @@
+import http from 'http';
+import url from 'url';
+
+import * as ShopifyErrors from '../error';
+import {Context} from '../context';
+
+/**
+ * Helper method to get the host URL for the app.
+ *
+ * @param request Current HTTP request
+ */
+export default function getHostAppUrl(request: http.IncomingMessage): string {
+  if (!request) {
+    throw new ShopifyErrors.MissingRequiredArgument(
+      'getHostAppUrl requires a request object argument',
+    );
+  }
+
+  if (!request.url) {
```

Comment:
Yeah, this really surprised me, but according to the type definitions it can be undefined:

<img width="447" alt="Screen Shot 2022-07-12 at 5 05 31 PM" src="https://user-images.githubusercontent.com/690791/178595270-f48f56cb-b17d-4aaf-b971-d5531e2ef345.png">

Unless I've mis-understood something?

Reply:
I thought url was always a string according to this https://nodejs.org/api/http.html#messageurl